src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { MdClose } from 'react-icons/md';
+import { bem } from '../../utils';
+import { Text } from '../Text';
+
+import styles from './Tag.scss';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** Value that helps us to understand if the component is currently selected or not. */
+    isSelected?: boolean;
+    /** Color is assigned to a background color of a wrapper */
+    bgColor?: string;
+    /** Max-width of a component */
+    maxWidth?: string;
+    /** Size of the text */
+    size?: 'small' | 'normal' | 'large';
+    /** Callback, that is fired when a user clicks on a delete icon */
+    onDelete?: () => void;
+    /** Callback, that is fired when a user clicks on an element */
+    onClick?: () => void;
+}
+
+const { block, elem } = bem('Tag', styles);
+
+export const Tag: React.FC<Props> = (props) => {
+    const { children, bgColor, maxWidth, size, onDelete, onClick, isSelected } = props;
+
+    const handleDeleteClick = (e: React.MouseEvent<HTMLElement>) => {
+        e.stopPropagation();
+
+        onDelete?.();
+    };
+
+    return (
+        <div
+            {...block({ className: isSelected && 'selected' })}
+            onClick={onClick}
+            onKeyDown={() => {}}
+            role="button"
+            tabIndex={0}
+            style={{
+                backgroundColor: bgColor,
+                maxWidth,
+            }}
+        >
+            <Text size={size} {...elem('Text', { elemClassName: styles.tagText })}>
```

Comment:
See the docs on the bem package. Elem should be called with something like `{...elem('tagText', prop)}`. maybe can omit props. The style definition should be `&__tagText {...}`

Reply:
Implemented.

Reply:
Implemented.